app/src/main/java/io/gnosis/safe/ui/settings/owner/ledger/LedgerController.kt

```diff
@@ -173,13 +173,19 @@ class LedgerController(val context: Context) {
 
     private var isScanning = false
 
-
-    fun startBleScan(fragment: Fragment, missingLocationPermissionHandler: () -> Unit) {
+    fun startBleScan(fragment: Fragment, missingPermissionHandler: () -> Unit) {
         if (!bluetoothAdapter.isEnabled) {
             promptEnableBluetooth(fragment)
         } else {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !context.hasPermission(Manifest.permission.ACCESS_FINE_LOCATION)) {
-                missingLocationPermissionHandler.invoke()
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
+                && Build.VERSION.SDK_INT < Build.VERSION_CODES.S
+                && (!context.hasPermission(Manifest.permission.ACCESS_FINE_LOCATION))
+            ) {
+                missingPermissionHandler.invoke()
+            } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S
```

Comment:
if and else if branches could be combined into one check because in both cases same statement is executed:
missingPermissionHandler.invoke()

Reply:
I would extract checking logic into a subroutine: checkPermissionsMissing(): Bool
if (checkPermissionsMissing) {
  missingPermissionHandler.invoke()
}

Reply:
even better to add this to the missingPermissionHandler as a method

Reply:
It could be added as a callback to the function signature also:
missingPermissionChecker: () -> Bool